app/models/operatingsystems/gentoo.rb

```diff
@@ -6,6 +6,7 @@ class << self
   end
 
   def mediumpath(host)
+    ''
   end
```

Comment:
You can remove this from Gentoo now, I haven't noticed there are more OSes which does not support it. Your if-statement will catch it.